source/common/http/conn_manager_impl.cc

```diff
@@ -730,6 +730,9 @@ void ConnectionManagerImpl::ActiveStream::encodeTrailers(ActiveStreamEncoderFilt
   }, this);
 #endif
 
+  if (buffered_response_data_ && buffered_response_data_->length() > 0) {
```

Comment:
Per offline convo with @fengli79 I am working on a proper solution to this part which unfortunately is pretty tricky. Hope to land this by the end of the week. It shouldn't change the rest of the code that much so we can continue to review/test in the interim.